README.md

```diff
@@ -412,6 +413,23 @@ list.map({ item => ... })
 
 ## <a name='lang'>Scala Language Features</a>
 
+### <a name='case_class_immutability'>Case Classes and Immutability</a>
+
+Case classes are regular classes but extended by the compiler to automatically support:
+- Public getters for constructor parameters
+- Copy constructor
+- Pattern matching on constructor parameters
+- Automatic toString/hash/equals implementation
+
+Constructor parameters should NOT be mutable for case classes.
+```scala
+// This is OK
+case class Person(name: String, age: Int)
+
+// This is NOT OK
```

Comment:
That does sound like a more general point, potentially exceeding a style guide, but we can at least give it as the reasoning. I thought there might also be a more Scala-y reason like you'll mess up all hash maps that contain this as a key :)